pkg/config/fsstore.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package config FSStore
+// implements file system KVStore interface
+package config
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+// FSStore Implements file system KVStore and change Store
+type FSStore struct {
+	// root directory
+	root string
+	// file suffix
+	suffix string
+
+	// testing and fault injection
+	writeCloserGetter writeCloserFunc
+	readfile          readFileFunc
+	mkdirAll          mkdirAllFunc
+	remove            removeFunc
+}
+
+// for testing
+type writeCloser interface {
+	io.WriteCloser
+	Name() string
+}
+type writeCloserFunc func(dir string, prefix string) (f writeCloser, err error)
+type readFileFunc func(filename string) ([]byte, error)
+type mkdirAllFunc func(path string, perm os.FileMode) error
```

Comment:
is it really necessary to abstract FS operations? is there any other implementation but using `io`?